app/Platform/Services/TriggerDiffService.php

```diff
@@ -0,0 +1,274 @@
+     *
+     * @param array<int, array<string, mixed>> $oldGroups decoded groups from the previous version
+     * @param array<int, array<string, mixed>> $newGroups decoded groups from the current version
+     * @return array{added: int, removed: int, modified: int}
+     */
+    public function computeSummary(array $oldGroups, array $newGroups): array
+    {
+        $diff = $this->computeDiff($oldGroups, $newGroups);
+
+        $numAdded = 0;
+        $numRemoved = 0;
+        $numModified = 0;
+
+        foreach ($diff as $group) {
+            foreach ($group['Conditions'] as $condition) {
+                switch ($condition['DiffStatus'] ?? 'unchanged') {
+                    case 'added':
+                        $numAdded++;
+                        break;
+                    case 'removed':
+                        $numRemoved++;
+                        break;
+                    case 'modified':
+                        $numModified++;
+                        break;
+                }
+            }
+        }
+
+        return ['added' => $numAdded, 'removed' => $numRemoved, 'modified' => $numModified];
+    }
+
+    /**
+     * Format the diff summary as a human-readable string.
+     *
+     * @param array{added: int, removed: int, modified: int} $summary
+     */
+    public function formatSummary(array $summary): string
+    {
+        $parts = [];
+
+        if ($summary['added'] > 0) {
+            $parts[] = $summary['added'] . ' added';
+        }
+        if ($summary['removed'] > 0) {
+            $parts[] = $summary['removed'] . ' removed';
+        }
+        if ($summary['modified'] > 0) {
+            $parts[] = $summary['modified'] . ' modified';
+        }
+
+        if (empty($parts)) {
+            return 'No changes';
+        }
+
+        return implode(', ', $parts);
+    }
+
+    /**
+     * Index groups by their label for easier lookup.
+     *
+     * @param array<int, array<string, mixed>> $groups
+     * @return array<string, array<string, mixed>>
+     */
+    private function indexGroupsByLabel(array $groups): array
+    {
+        $indexed = [];
+        foreach ($groups as $group) {
+            $indexed[$group['Label']] = $group;
+        }
+
+        return $indexed;
+    }
+
+    /**
+     * Diff two arrays of conditions.
+     *
+     * @param array<int, array<string, mixed>> $oldConditions
+     * @param array<int, array<string, mixed>> $newConditions
+     * @return array<int, array<string, mixed>> conditions with DiffStatus and RowIndex added
+     */
+    private function diffConditions(array $oldConditions, array $newConditions): array
+    {
+        // Compare conditions by position since order matters in triggers.
+        $result = [];
+        $maxLen = max(count($oldConditions), count($newConditions));
+
+        for ($i = 0; $i < $maxLen; $i++) {
+            $old = $oldConditions[$i] ?? null;
+            $new = $newConditions[$i] ?? null;
```

Comment:
I realize this algorithm is very naive, but it shouldn't be too hard to detect when a new condition is added in the middle of stuff. In this case, I added a new condition 3. Conditions 4 and 5 are identical to the old conditions 3 and 4.

<img width="1444" height="277" alt="Image" src="https://github.com/user-attachments/assets/848208c2-0129-4e58-8d30-991ed254514a" />

RATools goes one step farther and detects any moved conditions (keeping them in the position associated to the current version), but that gets significantly more complicated when you have to consider chains (AddAddress/AddSource/etc).